controllers/storagecluster/reconcile.go

```diff
@@ -838,6 +853,25 @@ func (r *StorageClusterReconciler) ownStorageConsumersInNamespace(instance *ocsv
 	return reconcile.Result{}, nil
 }
 
+func (r *StorageClusterReconciler) calculateBackwardCompatibilityMode(storageCluster *ocsv1.StorageCluster) (*util.BackwardCompatabilityInfo, error) {
+	if storageCluster.Spec.AllowRemoteStorageConsumers {
+		return &util.BackwardCompatabilityInfo{Pre4_19DeploymentMode: util.ProviderDeploymentMode}, nil
+	} else if storageCluster.Spec.ExternalStorage.Enable {
+		return &util.BackwardCompatabilityInfo{Pre4_19DeploymentMode: util.ExternalDeploymentMode}, nil
+	} else {
+		svgPre4_19 := &rookCephv1.CephFilesystemSubVolumeGroup{}
+		svgPre4_19.Name = fmt.Sprintf("%s-%s", util.GenerateNameForCephFilesystem(storageCluster.Name), subVolumeGroupName)
+		svgPre4_19.Namespace = storageCluster.Namespace
+		if err := r.Get(r.ctx, client.ObjectKeyFromObject(svgPre4_19), svgPre4_19); client.IgnoreNotFound(err) != nil {
```

Comment:
this heuristic is time dependent and when storageconsumer unit in cluster controller is reconciled, pre4.19 svg will be marked for deletion then in an upgraded internal cluster we always send `nil, nil`, is that what we want?